File: Problem_10/problem_10.erl
-module(problem_10).
-compile(export_all).

-import(math).

start()->
    io:format("result: ~p~n", [problem_10(2000000)]).

problem_10(Max) ->
    Sum = sum(Max, 2, 0),
    Sum.


is_prime(N, PFactor)->
    MaxFactor = math:sqrt(N),
    if PFactor =< MaxFactor ->
	    if  N rem PFactor == 0 ->
		    false;
		true ->
		    isprime(N, PFactor + 1)
	    end;
       true ->
	    true
    end.     

sum(Max, Start, Sum) ->
    if Start =< Max ->
	    Prime = is_prime(Start, 2),
	    if Prime == true ->
		    Newsum = Sum + Start,
		    sum(Max, Start + 1, Newsum);
	       true ->
		    sum(Max, Start + 1, Sum)
	    end;
       true ->
	    Sum
    end.

File: project-euler/Problem_10/problem_10.fs
open System

let is_prime (x:int64) =
    let rec check i =
        double i > sqrt (double x) || (x % i <> 0L && check (i + 1L))
    check 2L  

let problem_10 (n:int64) =
    seq { for i in 2L .. n do
          if is_prime i then yield i }
    |> Seq.sum

[<EntryPoint>]
let main(args : string[]) =
    printfn "result = %A" (problem_10 2000000L)
    0
